tests/basic.js

```diff
@@ -1,44 +1,113 @@
+import gentleRegisterProperty from "./util/gentle-register-property.js";
+import types from "./util/types.js";
+
+let dummy;
 
 export default {
 	name: "Basic",
 
-	run ({property, meta, initial, value}) {
-		let dummy = document.createElement("div");
+	map (value) {
+		if (this.data.property) {
+			dummy.style.setProperty(this.data.property, value);
+			let computed = getComputedStyle(dummy).getPropertyValue(this.data.property);
+			console.log(computed, value);
+			return computed;
+		}
+
+		return value;
+	},
+
+	beforeAll () {
+		dummy = document.createElement("div");
 		document.body.append(dummy);
+	},
 
-		let observer;
-		return new Promise((resolve, reject) => {
-			observer = new StyleObserver(records => {
-				resolve(records);
-			});
-
-			if (meta) {
-				gentleRegisterProperty(property, meta);
-			}
-
-			if (initial) {
-				dummy.style.setProperty(property, initial);
-			}
-
-			observer.observe(dummy, property);
-
-			dummy.style.setProperty(property, value);
-
-			// Timeout after 500ms
-			setTimeout(() => reject(), 500);
-		})
-		.catch(_ => {
-			return [{ value: "Timed out" }];
-		})
-		.then(records => records[0].value)
-		.finally(() => {
-			observer.unobserve(dummy);
-			dummy.remove();
+	beforeEach () {
+		let element = document.createElement("div");
+		document.body.append(element);
+
+		let result = {};
+		result.promise = new Promise((resolve, reject) => {
+			result.resolve = resolve;
+			result.reject = reject;
+		});
+
+		let observer = new StyleObserver(records => {
+			let value = records[0].value;
+			result.resolve(value);
 		});
+
+		Object.assign(this.data, { element, observer, result });
 	},
 
-	tests,
+	run (property, initialValue, value) {
+		let { observer, element, result } = this.data;
+
+		element.style.setProperty(property, initialValue);
+		observer.observe(element, property);
+		element.style.setProperty(property, value);
+
+		setTimeout(() => result.reject("Timeout"), 500);
+
+		return result.promise;
+	},
+
+	afterEach () {
+		this.data.observer.unobserve(this.data.element);
+		this.data.element.remove();
+	},
+
+	tests: [
+		{
+			name: "Built-in properties",
+			tests: Object.entries(types).map(([id, meta]) => {
+				return {
+					data: meta,
+					args: [meta.property, meta.values[0], meta.initialValue],
+					expect: meta.values[0],
+				};
+			}),
+		},
+		{
+			name: "Registered custom properties",
+			getName () {
+				return this.data.syntax;
+			},
+			tests: Object.entries(types).map(([id, meta]) => {
+				let property = `--registered-${meta.id}`;
+				return {
+					beforeEach () {
+						let definition = {
+							name: `--registered-${meta.id}`,
+							syntax: meta.syntax,
+							initialValue: meta.initialValue,
+						};
+						try {
+							gentleRegisterProperty(property, meta);
+						}
+						catch (e) {
+							// Unsupported syntax?
+							console.log(e, definition);
+							this.skip = true;
+						}
+
+						// FIXME Why did I need two .parent here?
```

Comment:
You didn't. It works with one parent (I've checked). Is it possible that it didn't work because of some other issue?